docs/DeveloperGuide.md

```diff
@@ -378,234 +375,353 @@ Use case ends.
 
 
 * 2b. SellSavvy detects that the new customer has tags with similar names.
-  * 2b1. SellSavvy gives a warning that there are similar tags in the new customer.
+  * 2b1. SellSavvy gives a warning that there are similar tags on the new customer.
   
   Use case resumes from step 3.
 
 **Use case 3: Delete Customer and All Orders Related to The Customer**
 
-* **System**: SellSavvy
-* **Actor**: User
 * **Use Case**: UC03 - Delete Customer and All Orders related to the customer
 * **Preconditions**: There are customers displayed in the customer list.
 * **Guarantees**:
     * Customer and all their orders will be deleted if input parameters are valid.
 
 **MSS**
 
-1. User wants to delete a customer.
-2. User finds the customer index from the list.
-3. User deletes the customer by their index.
-4. SellSavvy updates the displayed list of customers and indicates that delete is successful.
+1. User finds the customer index from the list.
+2. User deletes the customer by their index.
+3. SellSavvy updates the displayed list of customers and indicates that delete is successful.
 
 Use case ends.
 
 **Extensions**
 
-* 3a. SellSavvy detects that there are no customers with the specified index.
-  * 3a1. SellSavvy displays an error that the customer index is invalid. 
-  
-  Use case ends.
+* 2a. SellSavvy detects customer index is missing or non-positive.
+    * 2a1. SellSavvy displays an error about the format and states the command format.
 
+  Use case ends.
 
-* 3b. SellSavvy detects that there are no customers with the specified index.
-    * 3a1. SellSavvy displays an error that the customer index is invalid.
 
+* 2b. SellSavvy detects that there are no customers with the specified index.
+  * 2b1. SellSavvy displays an error that the customer index is invalid. 
+  
   Use case ends.
 
 **Use case 4: Find the Customer by their Name**
 
-* **System**: SellSavvy
-* **Actor**: User
 * **Use Case**: UC04 - Find the Customer by their Name
 * **Guarantees**:
   * All customers displayed will have at least one of the specified keywords in their name.
 
 **MSS**
 
-1. User wants to search for a specific customer.
-2. User specifies keyword(s) of the name of the customer
-3. SellSavvy displays all customers whose names have at least one of the keywords.
+1. User specifies keyword(s) of the name of the customer(s) they want to find.
+2. SellSavvy displays all customers whose names have at least one of the keywords.
```

Comment:
keyword(s) ?

Reply:
There can be multiple keywords in the parameter.

The implementation is a little weird as well since if you state multiple keywords, the result customers only need to contain at least one of the keywords, so I'm not 100% sure how to phrase this part since you can return multiple customers with completely different names if you put multiple keywords.